test/EntityFramework6.Npgsql.Tests/PatternMatchingTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2016 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using Npgsql;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Data.Entity;
+using System.Linq;
+using System.Text;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Data.Entity.Core.Metadata.Edm;
+using System.Data.Entity.Core.Objects;
+using System.Data.Entity.Infrastructure;
+using NpgsqlTypes;
+using System.Text.RegularExpressions;
+
+namespace EntityFramework6.Npgsql.Tests
+{
+    class PatternMatchingTests : EntityFrameworkTestBase
+    {
+        [Test]
+        [TestCase("blog", "blog", "BLOG", TestName = "MatchRegex case-sensitive")]
+        [TestCase("^blog$", "blog", "some \nblog\n name", TestName = "MatchRegex ^ and $ match beginning and end")]
+        [TestCase("some .* name", "some blog name", "some \n name", TestName = "MatchRegex . matches all except \\n")]
+        [TestCase("some blog name", "some blog name", "someblogname", TestName = "MatchRegex whitespace not ignored in pattern")]
+        public void MatchRegex(string pattern, string matchingInput, string mismatchingInput)
```

Comment:
You can remove MatchRegex from test TestName, in the unit test runner the test names will appear under the method name so no need to mention it again in the test name.